test/common/router/rds_impl_test.cc

```diff
@@ -613,6 +613,95 @@ name: foo_route_config
             route_config_provider_manager_->dumpRouteConfigs()->dynamic_route_configs().size());
 }
 
+TEST_F(RouteConfigProviderManagerImplTest, Delta) {
+  Buffer::OwnedImpl data;
+
+  // Get a RouteConfigProvider. This one should create an entry in the RouteConfigProviderManager.
+  setup();
+
+  EXPECT_FALSE(provider_->configInfo().has_value());
+
+  Protobuf::RepeatedPtrField<ProtobufWkt::Any> route_configs;
+  route_configs.Add()->PackFrom(parseRouteConfigurationFromV2Yaml(R"EOF(
+name: foo_route_config
+virtual_hosts:
+  - name: bar
+    domains: ["*"]
+    routes:
+      - match: { prefix: "/" }
+        route: { cluster: baz }
+)EOF"));
+
+  RdsRouteConfigSubscription& subscription =
+      dynamic_cast<RdsRouteConfigProviderImpl&>(*provider_).subscription();
+
+  Protobuf::RepeatedPtrField<envoy::api::v2::Resource> resources;
+  auto* resource = resources.Add();
+  *resource->mutable_resource() = route_configs[0];
+  resource->set_version("1");
+  subscription.onConfigUpdate(resources, {}, "1");
+
+  RouteConfigProviderPtr provider2 = route_config_provider_manager_->createRdsRouteConfigProvider(
+      rds_, factory_context_, "foo_prefix");
+
+  // provider2 should have route config immediately after create
+  EXPECT_TRUE(provider2->configInfo().has_value());
+
+  // So this means that both provider have same subscription.
+  EXPECT_EQ(&dynamic_cast<RdsRouteConfigProviderImpl&>(*provider_).subscription(),
+            &dynamic_cast<RdsRouteConfigProviderImpl&>(*provider2).subscription());
+  EXPECT_EQ(&provider_->configInfo().value().config_, &provider2->configInfo().value().config_);
+
+  std::string config_json2 = R"EOF(
+    {
+      "api_type": "REST",
```

Comment:
How does a REST RDS configuration figure into delta xDS testing?

Reply:
Won't accept DELTA_GRPC. Maybe I should just delete this test.